inference-platforms/agent.py

```diff
@@ -0,0 +1,68 @@
+# run like this: uv run --exact -q --env-file .env agent.py
+# /// script
+# dependencies = [
+#     "openai-agents",
+#     "httpx",
+#     "mcp",
+#     "elastic-opentelemetry",
+#     "openinference-instrumentation-openai-agents",
+#     "opentelemetry-instrumentation-httpx",
+#     "openinference-instrumentation-mcp",
+# ]
+# ///
+# ruff: noqa: E402
+from opentelemetry.instrumentation import auto_instrumentation
+
+# This must precede any other imports you want to instrument!
+auto_instrumentation.initialize()
+
+import asyncio
+import os
+from datetime import datetime, timedelta
+
+from agents import (
+    Agent,
+    OpenAIProvider,
+    RunConfig,
+    Runner,
+    Tool,
+)
+from agents.mcp import MCPServerStreamableHttp, MCPUtil
+
+
+async def run_agent(tools: list[Tool]):
+    model_name = os.getenv("AGENT_MODEL", "gpt-5-nano")
+    model = OpenAIProvider(use_responses=False).get_model(model_name)
+    agent = Agent(
+        name="flight-search-agent",
+        model=model,
+        tools=tools,
+    )
+
+    next_week = (datetime.now() + timedelta(weeks=1)).strftime("%Y-%m-%d")
+    result = await Runner.run(
+        starting_agent=agent,
+        input=f"Give me the best flight from New York to Kota Kinabalu on {next_week}",
+        run_config=RunConfig(workflow_name="flight search"),
+    )
+    print(result.final_output)
+
+
+async def main():
+    mcp_url = os.getenv("MCP_URL", "https://mcp.kiwi.com")
+    async with MCPServerStreamableHttp(
+        {
+            "url": mcp_url,
+            "timeout": 30.0,
+        },
+        cache_tools_list=True,
```

Comment:
Just a yolo comment since I didn't check it at all, but it wouldn't surprise me if this enables the background polling you saw